### PR TITLE
Add test for persistent cache clean up after relocation

### PR DESCRIPTION
**Pull request by tlrx**
This pull request adds a test to verify that the persistent cache is correctly cleaned up after a shard is relocated. I've been surprised to see we don't have integ tests for this.

### CODE REVIEW

**Review by fcofdez — APPROVED** [review]
LGTM 👍. Thanks for adding the test!

**Review by original-brownbear — APPROVED** [review]
LGTM2 sorry forgot to click approve yesterday :)

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/full/SearchableSnapshotsPersistentCacheIntegTests.java [hunk 1]
```diff
@@ -154,4 +161,113 @@ public Settings onNodeStopped(String nodeName) {
...
+        createAndPopulateIndex(
+            indexName,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 3)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, randomIntBetween(0, 1))
+        );
+
+        final String snapshotName = prefix + "snapshot";
+        final SnapshotInfo snapshotInfo = createFullSnapshot(fsRepoName, snapshotName);
+        assertThat(snapshotInfo.successfulShards(), greaterThan(0));
```
[the remark this answers is not in the document]
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/full/SearchableSnapshotsPersistentCacheIntegTests.java]: +1, thanks

**Comment by tlrx** [comment]
Thanks Francisco and Armin!

**Pull request merged**